### DISCUSSION TITLE
Fast refresh only works when a file only export components. Move your component(s) to a separate file.eslint(react-refresh/only-expor

### DISCUSSION THREAD

**Discussion by vaynevayne** [remix-run/react-router#10969] [Q&A]
How can I solve this eslint problem?

```jsx
// router.tsx
const AdAccount = lazy(() => import('@/pages/AdAccount'))

```

```jsx
// AdAccount 
const AdAccount = null
export default AdAccount
```

<img width="705" alt="image" src="https://github.com/remix-run/react-router/assets/124666577/6fc0e3be-9b8a-43b7-ab27-77210c5ffa37">

**Comment by riadkhan60** [comment]
If you read the article linked below you might understand the issue.

https://nextjs.org/docs/architecture/fast-refresh

They have introduced a new method of refreshing a file that aims to be faster. However, for this method to work effectively, the file should only contain components, not constant values or functions. This isn't something that could be called an error; it's completely acceptable. All you need to do is separate the functions into another file, and that will resolve the issue
